server/store.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-plugin-zoom/server/zoom"
+)
+
+const (
+	postMeetingKey     = "post_meeting_"
+	zoomStateKeyPrefix = "zoomuserstate"
+	zoomUserByMMID     = "zoomtoken_"
+	zoomUserByZoomID   = "zoomtokenbyzoomid_"
+
+	meetingPostIDTTL  = 60 * 60 * 24 // One day
+	oAuthUserStateTTL = 60 * 5       // 5 minutes
+)
+
+func (p *Plugin) storeOAuthUserInfo(info *zoom.OAuthUserInfo) error {
+	config := p.getConfiguration()
+
+	encryptedToken, err := encrypt([]byte(config.EncryptionKey), info.OAuthToken.AccessToken)
+	if err != nil {
+		return errors.Wrap(err, "could not encrypt OAuth token")
+	}
+	info.OAuthToken.AccessToken = encryptedToken
+
+	encoded, err := json.Marshal(info)
+	if err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(zoomUserByMMID+info.UserID, encoded); err != nil {
+		return err
+	}
+
+	if err := p.API.KVSet(zoomUserByZoomID+info.ZoomID, encoded); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) fetchOAuthUserInfo(tokenKey, userID string) (*zoom.OAuthUserInfo, error) {
+	encoded, appErr := p.API.KVGet(tokenKey + userID)
+	if appErr != nil || encoded == nil {
+		return nil, errors.New("must connect user account to Zoom first")
+	}
+
+	var info zoom.OAuthUserInfo
+	if err := json.Unmarshal(encoded, &info); err != nil {
+		return nil, errors.New("could not to parse OAauth access token")
+	}
+
+	return &info, nil
+}
+
+func (p *Plugin) disconnectOAuthUser(userID string) error {
+	encoded, err := p.API.KVGet(zoomUserByMMID + userID)
+	if err != nil {
+		return errors.Wrap(err, "could not find OAuth user info")
+	}
+
+	var info zoom.OAuthUserInfo
+	if err := json.Unmarshal(encoded, &info); err != nil {
+		return errors.Wrap(err, "could not decode OAuth user info")
+	}
+
+	errByMattermostID := p.API.KVDelete(zoomUserByMMID + userID)
+	errByZoomID := p.API.KVDelete(zoomUserByZoomID + info.ZoomID)
+	if errByMattermostID != nil {
+		return errByMattermostID
+	}
+	if errByZoomID != nil {
+		return errByZoomID
+	}
```

Comment:
Nit pick: Using separate error variables is not idiomatic. 
```suggestion
	appErr := p.API.KVDelete(zoomUserByMMID + userID)
	if appErr != nil {
		return appErr
	}
	
	appErr = p.API.KVDelete(zoomUserByZoomID + info.ZoomID)
	if appErr != nil {
		return appErr
	}
```

Reply:
@hanzei In fact, I moved this piece of code from another file. I was going to update it exactly as you suggest, but then I thought maybe the original author wanted to make sure that the second `KVDelete` call gets to run even if the first one fails. 

What are your thoughts on this? I can apply your suggestion if you think the above concern is not important.

Reply:
If both calls to `KVDelete` fail, only one error will get logged. That would make debugging harder. Also, a dead record in the DB is not the end of the world.

Let's cleanup this code.`